pkg/actions/lua/databricks/client.go

```diff
@@ -0,0 +1,177 @@
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/databricks-sdk-go/service/sql"
+)
+
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-identifiers.html
+var identifierRegex = regexp.MustCompile(`\W`)
+
+var (
+	ErrInvalidTableName     = errors.New("invalid table name")
+	ErrTableCreationFailure = errors.New("failed creating table")
+)
+
+type Client struct {
+	workspaceClient *databricks.WorkspaceClient
+	ctx             context.Context
+}
+
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-identifiers.html
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html
+func validateTableName(tableName string) error {
+	if identifierRegex.Match([]byte(tableName)) {
+		return ErrInvalidTableName
+	} else if len(tableName) > 255 {
+		return ErrInvalidTableName
+	}
+	return nil
+}
+
+func validateTableLocation(tableLocation string) error {
+	_, err := url.ParseRequestURI(tableLocation)
+	return err
+}
+
+func validateTableInput(tableName, location string) error {
+	errName := validateTableName(tableName)
+	errLocation := validateTableLocation(location)
+	return errors.Join(errName, errLocation)
+}
+
+func (client *Client) createExternalTable(warehouseID, catalogName, schemaName, tableName, location string) (string, error) {
+	if err := validateTableInput(tableName, location); err != nil {
+		return "", errors.Join(ErrTableCreationFailure, err)
+	}
+	statement := fmt.Sprintf(`CREATE EXTERNAL TABLE %s LOCATION '%s'`, tableName, location)
+	esr, err := client.workspaceClient.StatementExecution.ExecuteAndWait(client.ctx, sql.ExecuteStatementRequest{
+		WarehouseId: warehouseID,
+		Catalog:     catalogName,
+		Schema:      schemaName,
+		Statement:   statement,
+	})
+	if err != nil {
+		return "", fmt.Errorf("external table creation failed: %w", err)
+	}
+	return esr.Status.State.String(), nil
+}
+
+func tableFullName(catalogName, schemaName, tableName string) string {
+	return fmt.Sprintf("%s.%s.%s", catalogName, schemaName, tableName)
+}
+
+func (client *Client) deleteTable(catalogName, schemaName, tableName string) error {
+	err := client.workspaceClient.Tables.DeleteByFullName(client.ctx, tableFullName(catalogName, schemaName, tableName))
+	if err != nil {
+		return fmt.Errorf("failed deleting an existing table: %w", err)
+	}
+	return nil
+}
+
+func (client *Client) createOrGetSchema(catalogName, schemaName string) (*catalog.SchemaInfo, error) {
+	schemaInfo, err := client.workspaceClient.Schemas.Create(client.ctx, catalog.CreateSchema{
+		Name:        schemaName,
+		CatalogName: catalogName,
+	})
+
+	if err == nil {
+		return schemaInfo, nil
+	}
+	if alreadyExists(err) {
+		// Full name of schema, in form of <catalog_name>.<schema_name>
+		schemaInfo, err = client.workspaceClient.Schemas.GetByFullName(client.ctx, catalogName+"."+schemaName)
+		if err == nil {
+			return schemaInfo, nil
+		}
+	}
+	return nil, err
```

Comment:
The comment here to show that in this case when `GetByFullName` fails we report the original error - related to the previous comment.